aten/src/ATen/native/cuda/GridSampler.cu

```diff
@@ -15,40 +15,39 @@ using at::native::detail::GridSamplerInterpolation;
 using at::native::detail::GridSamplerPadding;
 
 namespace {
-  template <typename scalar_t>
+  template <typename scalar_t, typename index_t>
```

Comment:
> The decision is: a) Include a 64-bit indexed kernel (increasing binary size), or b) raise an error when 32-bit indexing would overflow.

Regarding the decision, Natalia suggested that this won't increase the binary size very much (unless you have data otherwise) https://github.com/pytorch/pytorch/issues/41656#issuecomment-665781990 and that we should implement 64 bit indexing if we can. Since this PR already implements 64 bit indexing for CUDA I'm in favor of keeping it.